src/common/snippets/src/op/subgraph.cpp

```diff
@@ -510,7 +509,11 @@ void Subgraph::control_flow_transformations(size_t min_parallel_work_amount, siz
     //       (this might happen if tail loop and main loop have different increments)
     //    3. OptimizeLoopSingleEvaluation must be called after CleanupLoopOffsets
     //       since CleanupLoopOffsets can't handle loops with evaluate_once = true
-    gen_pipeline.register_pass<lowered::pass::AssignRegisters>(reg_type_mapper, get_generator()->get_target_machine()->get_reg_count());
+
+    lowered::RegManager reg_manager(get_generator());
+    gen_pipeline.register_pass<lowered::pass::InitLiveRanges>(reg_manager);
+    gen_pipeline.register_pass<lowered::pass::AssignRegisters>(reg_manager);
+    gen_pipeline.register_pass<lowered::pass::InsertRegSpills>(reg_manager);
```

Comment:
What do you think about to change order of register-related passes and `InsertSpecificIteration`? As far as I remember, we just need to support `Result` with several inputs to connect with decomposed loop iterations.

I think about it because in `InitLiveRanges` we set live ranges of the current LinearIR state. After `InsertSpecificIterations`, these values might be invalid - we inserted new expressions with new execution orders but not update live ranges - they're valid for the original expressions.
Probably it might leads to problems in the future.

Reply:
You are right, in theory `InsertSpecificIterations` might break register allocation if some of the handlers does smth nasty. We can try to change the order now, since the refactored passes are much more generic and it might even work out of the box. However it would significantly impact the compilation time because register reuse is global and relatively expensive optimization.
I think we should weight all pros and cons carefully before doing that. Probably it would be more benefitial to restrict allowed transformations inside handlers than to do reg assignment on expanded LIR.